imqsauth/http.go

```diff
@@ -770,3 +837,65 @@ func httpHandlerGetGroups(central *ImqsCentral, w http.ResponseWriter, r *httpRe
 		httpSendGroupsJson(w, groups)
 	}
 }
+
+func (central *ImqsCentral) handleWebSocket(ws *websocket.Conn) {
+	permOK := false
+	closeConnection := false
+	if token, err := authaus.HttpHandlerPrelude(&central.Config.Authaus.HTTP, central.Central, ws.Request()); err == nil {
+		if permList, errDecodePerms := authaus.PermitResolveToList(token.Permit.Roles, central.Central.GetRoleGroupDB()); errDecodePerms != nil {
+			closeConnection = true
+		} else {
+			if !permList.Has(PermEnabled) {
+				closeConnection = true
+			} else {
+				permOK = true
+			}
+		}
+	} else {
+		closeConnection = true
+	}
+
+	if !permOK {
+		closeConnection = true
+	}
+
+	if closeConnection {
+		ws.Close()
+		return
+	}
+
+	id := central.Add(ws)
+
+	central.Central.Log.Infof("Added auth web socket client %d", id)
+	var msg string
+	for {
+		// Currently we should not receive anything on this websocket, should we remove this code entirely?
+		err := websocket.Message.Receive(ws, &msg)
+
+		if (err != nil) && (err != io.EOF) {
+			central.Central.Log.Infof("handleWebSocket: EOF or Error on websocket %d: %v", id, err)
+			ws.Close()
+			delete(central.WebSocketConnections, id)
+			central.Central.Log.Infof("handleWebSocket: Web socket connection removed %d", id)
+			return
+		}
+	}
+}
+
+func (central *ImqsCentral) sendAll(msg string) {
```

Comment:
This function needs a better name, such as "broadcastToAllListeners".
Under what conditions does websocket.Message.Send return io.EOF?